File: day-before.py
#!/usr/bin/env python

# given YYYY-MM-DD return the one day earlier as YYYYMMDD

import sys, os
from datetime import date, timedelta

if len(sys.argv) != 2:
    print "Usage: ", sys.argv[0], " YYYY-MM-DD"
    sys.exit(os.EX_USAGE)

# convert input string to date, subtract one day
#
inarray = sys.argv[1].split('-')
log_day = date(int(inarray[0]), int(inarray[1]), int(inarray[2]))
merge_day = log_day - timedelta(1)

# convert back to string and print
print merge_day.strftime("%Y%m%d")

sys.exit()

File: day-before.rb
#!/usr/bin/env ruby -w

# given YYYY-MM-DD return the one day earlier as YYYYMMDD

if ARGV.empty?
  print "Usage: ", $0, " YYYY-MM-DD\n"
  exit 1
end
inarray = ARGV[0].split('-')

# convert to seconds since epoch, subtract one day (86400 seconds)
#
log_day = Time.gm(inarray[0], inarray[1], inarray[2])
merge_day = log_day - 86400

# convert back to string and get bits we need
printf("%d%02d%02d\n", merge_day.year, merge_day.month, merge_day.day)

exit 0
